### ISSUE TITLE
[BUG] QueueClient.SendMessageAsync sends bad XML request and corrupts client state with non-ASCII characters

### ISSUE THREAD

**Issue by joelverhagen** [Azure/azure-sdk-for-net#19886]
**Describe the bug**
I have a queue message that contains the `ä` character. This caused a `RequestFailedException` **(problem 1)** due to the XML request body being truncated, e.g. the request body looks like:
```xml
<QueueMessage><MessageText>ä</MessageText></QueueMessage
```
(note the missing `>` at the end)

Then, any next message (even if it is all ASCII characters) fails with an `XmlException` **(problem 2)** even though the server returns an HTTP 400 about an invalid request verb. The beginning of the request of this second request looks like:
```
>POST https://REDACTED.queue.core.windows.net/test/messages HTTP/1.1
Host: REDACTED.queue.core.windows.net
```
(note the `>` at the beginning, I guess we found the missing `>` 😄)

I should note that the second exception thrown seems to be related to the SDK freaking out due to an HTML error response. This hides the actual error response which is visible in Fiddler:

```
HTTP/1.1 400 Bad Request
Content-Type: text/html; charset=us-ascii
Server: Microsoft-HTTPAPI/2.0
Date: Sat, 27 Mar 2021 18:15:59 GMT
Connection: close
Content-Length: 326

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN""http://www.w3.org/TR/html4/strict.dtd">
<HTML><HEAD><TITLE>Bad Request</TITLE>
<META HTTP-EQUIV="Content-Type" Content="text/html; charset=us-ascii"></HEAD>
<BODY><h2>Bad Request - Invalid Verb</h2>
<hr><p>HTTP Error 400. The request verb is invalid.</p>
</BODY></HTML>
```

This issue seems similar to to https://github.com/Azure/azure-sdk-for-net/issues/15478 but my issue about the `SendMessageAsync` side and has to do with corrupting the client state such that any subsequent `SendMessageAsync` fails.

**Expected behavior**
I should be able to send messages with non-ASCII characters, as long as they are properly encoded. `WindowsAzure.Storage` SDK works fine here.

Errors in a previous `SendMessageAsync` should not put the client in a corrupted state and subsequent, ASCII `SendsMessageAsync` invocations should succeed immediately.

**Actual behavior (include Exception or Stack Trace)**
Exception from a non-ASCII message:
```
Azure.RequestFailedException: XML specified is not syntactically valid.
RequestId:c06972d2-3003-002b-6935-23661c000000
Time:2021-03-27T18:15:59.2421247Z
Status: 400 (XML specified is not syntactically valid.)
ErrorCode: InvalidXmlDocument

Additional Information:
LineNumber: 1
LinePosition: 57
Reason: Error parsing Xml content

Headers:
Server: Windows-Azure-Queue/1.0,Microsoft-HTTPAPI/2.0
x-ms-request-id: c06972d2-3003-002b-6935-23661c000000
x-ms-version: 2018-11-09
x-ms-error-code: InvalidXmlDocument
Date: Sat, 27 Mar 2021 18:15:59 GMT
Content-Length: 327
Content-Type: application/xml

   at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync_CreateResponse(ClientDiagnostics clientDiagnostics, Response response)
   at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri resourceUri, QueueSendMessage message, String version, Nullable`1 visibilitytimeout, Nullable`1 messageTimeToLive, Nullable`1 timeout, String requestId, Boolean async, String operationName, CancellationToken cancellationToken)
   at Azure.Storage.Queues.QueueClient.SendMessageInternal(BinaryData message, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, Boolean async, CancellationToken cancellationToken)
   at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, CancellationToken cancellationToken)
   at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText)
   at ConsoleApp6.Program.NewSdkAsync(String connectionString, String[] messages) in C:\Users\joelv\Desktop\ConsoleApp6\ConsoleApp6\Program.cs:line 52
```

Subsequent exception from corrupted state.

```
System.Xml.XmlException: '"' is an unexpected token. Expecting whitespace. Line 1, position 50.
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.DtdParserProxy.System.Xml.IDtdParserAdapter.Throw(Exception e)
   at System.Xml.DtdParser.Throw(String res, String arg, Int32 lineNo, Int32 linePos)
   at System.Xml.DtdParser.ParseExternalId(Token idTokenType, Token declType, String& publicId, String& systemId)
   at System.Xml.DtdParser.ParseInDocumentDtd(Boolean saveInternalSubset)
   at System.Xml.DtdParser.Parse(Boolean saveInternalSubset)
   at System.Xml.DtdParser.System.Xml.IDtdParser.ParseInternalDtd(IDtdParserAdapter adapter, Boolean saveInternalSubset)
   at System.Xml.XmlTextReaderImpl.ParseDtd()
   at System.Xml.XmlTextReaderImpl.ParseDoctypeDecl()
   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
   at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
   at System.Xml.Linq.XDocument.Load(Stream stream, LoadOptions options)
   at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync_CreateResponse(ClientDiagnostics clientDiagnostics, Response response)
   at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri resourceUri, QueueSendMessage message, String version, Nullable`1 visibilitytimeout, Nullable`1 messageTimeToLive, Nullable`1 timeout, String requestId, Boolean async, String operationName, CancellationToken cancellationToken)
   at Azure.Storage.Queues.QueueClient.SendMessageInternal(BinaryData message, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, Boolean async, CancellationToken cancellationToken)
   at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, CancellationToken cancellationToken)
   at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText)
   at ConsoleApp6.Program.NewSdkAsync(String connectionString, String[] messages) in C:\Users\joelv\Desktop\ConsoleApp6\ConsoleApp6\Program.cs:line 52System.Xml.XmlException: '"' is an unexpected token. Expecting whitespace. Line 1, position 50.
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.DtdParserProxy.System.Xml.IDtdParserAdapter.Throw(Exception e)
   at System.Xml.DtdParser.Throw(String res, String arg, Int32 lineNo, Int32 linePos)
   at System.Xml.DtdParser.ParseExternalId(Token idTokenType, Token declType, String& publicId, String& systemId)
   at System.Xml.DtdParser.ParseInDocumentDtd(Boolean saveInternalSubset)
   at System.Xml.DtdParser.Parse(Boolean saveInternalSubset)
   at System.Xml.DtdParser.System.Xml.IDtdParser.ParseInternalDtd(IDtdParserAdapter adapter, Boolean saveInternalSubset)
   at System.Xml.XmlTextReaderImpl.ParseDtd()
   at System.Xml.XmlTextReaderImpl.ParseDoctypeDecl()
   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
   at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
   at System.Xml.Linq.XDocument.Load(Stream stream, LoadOptions options)
   at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync_CreateResponse(ClientDiagnostics clientDiagnostics, Response response)
   at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri resourceUri, QueueSendMessage message, String version, Nullable`1 visibilitytimeout, Nullable`1 messageTimeToLive, Nullable`1 timeout, String requestId, Boolean async, String operationName, CancellationToken cancellationToken)
   at Azure.Storage.Queues.QueueClient.SendMessageInternal(BinaryData message, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, Boolean async, CancellationToken cancellationToken)
   at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, CancellationToken cancellationToken)
   at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText)
   at ConsoleApp6.Program.NewSdkAsync(String connectionString, String[] messages) in C:\Users\joelv\Desktop\ConsoleApp6\ConsoleApp6\Program.cs:line 52
```

**To Reproduce**

Here is a simple C# program that shows the problem. It demonstrates that the old SDK has no problem with the same message. Note that I am not using base64 encoding anywhere.

```csharp
using System;
using System.Threading.Tasks;
using Azure.Storage.Queues;
using Microsoft.WindowsAzure.Storage;
using Microsoft.WindowsAzure.Storage.Queue;

namespace ConsoleApp6
{
    class Program
    {
        static async Task Main(string[] args)
        {
            var connectionString = "UseDevelopmentStorage=true";
            var messages = new[] { "ä", "ë", "ï", "ö", "ü", "a", "e", "i", "o", "u" };
            await OldSdkAsync(connectionString, messages);
            await NewSdkAsync(connectionString, messages);
        }

        private static async Task OldSdkAsync(string connectionString, string[] messages)
        {
            var client = CloudStorageAccount.Parse(connectionString);
            var queue = client.CreateCloudQueueClient().GetQueueReference("test");
            queue.EncodeMessage = false;
            await queue.CreateIfNotExistsAsync();
            foreach (var message in messages)
            {
                try
                {
                    Console.Write("Old SDK: " + message + " = ");
                    await queue.AddMessageAsync(new CloudQueueMessage(message));
                    Console.WriteLine("success");
                }
                catch (Exception ex)
                {
                    Console.WriteLine(ex.GetType().Name);
                }
                Console.WriteLine(new string('=', 60));
            }
        }

        private static async Task NewSdkAsync(string connectionString, string[] messages)
        {
            var client = new QueueServiceClient(connectionString);
            var queue = client.GetQueueClient("test");
            await queue.CreateIfNotExistsAsync();
            foreach (var message in messages)
            {
                try
                {
                    Console.Write("New SDK: " + message + " = ");
                    await queue.SendMessageAsync(message);
                    Console.WriteLine("success");
                }
                catch (Exception ex)
                {
                    Console.WriteLine("error");
                    Console.WriteLine(ex);
                }
                Console.WriteLine(new string('=', 60));
            }
        }
    }
}
````

<details>
<summary>stdout from runnning this repro code</summary>

    Old SDK: ä = success
    ============================================================
    Old SDK: ë = success
    ============================================================
    Old SDK: ï = success
    ============================================================
    Old SDK: ö = success
    ============================================================
    Old SDK: ü = success
    ============================================================
    Old SDK: a = success
    ============================================================
    Old SDK: e = success
    ============================================================
    Old SDK: i = success
    ============================================================
    Old SDK: o = success
    ============================================================
    Old SDK: u = success
    ============================================================
    New SDK: ä = error
    Azure.RequestFailedException: XML specified is not syntactically valid.
    RequestId:be91c085-7ffb-4eab-bb82-5f2bcd8ff4ff
    Time:2021-03-27T18:28:39.3883799Z
    Status: 400 (XML specified is not syntactically valid.)
    ErrorCode: InvalidXmlDocument

    Additional Information:
    LineNumber: 1
    LinePosition: 57
    Reason: Error parsing Xml content

    Headers:
    Server: Windows-Azure-Queue/1.0,Microsoft-HTTPAPI/2.0
    x-ms-request-id: be91c085-7ffb-4eab-bb82-5f2bcd8ff4ff
    x-ms-version: 2018-11-09
    x-ms-error-code: InvalidXmlDocument
    Date: Sat, 27 Mar 2021 18:28:39 GMT
    Content-Length: 327
    Content-Type: application/xml

    at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync_CreateResponse(ClientDiagnostics clientDiagnostics, Response response)
    at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri resourceUri, QueueSendMessage message, String version, Nullable`1 visibilitytimeout, Nullable`1 messageTimeToLive, Nullable`1 timeout, String requestId, Boolean async, String operationName, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageInternal(BinaryData message, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, Boolean async, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText)
    at ConsoleApp6.Program.NewSdkAsync(String connectionString, String[] messages) in C:\Users\joelv\Desktop\ConsoleApp6\ConsoleApp6\Program.cs:line 53
    ============================================================
    New SDK: ë = error
    System.Xml.XmlException: '"' is an unexpected token. Expecting whitespace. Line 1, position 50.
    at System.Xml.XmlTextReaderImpl.Throw(Exception e)
    at System.Xml.XmlTextReaderImpl.DtdParserProxy.System.Xml.IDtdParserAdapter.Throw(Exception e)
    at System.Xml.DtdParser.Throw(String res, String arg, Int32 lineNo, Int32 linePos)
    at System.Xml.DtdParser.ParseExternalId(Token idTokenType, Token declType, String& publicId, String& systemId)
    at System.Xml.DtdParser.ParseInDocumentDtd(Boolean saveInternalSubset)
    at System.Xml.DtdParser.Parse(Boolean saveInternalSubset)
    at System.Xml.DtdParser.System.Xml.IDtdParser.ParseInternalDtd(IDtdParserAdapter adapter, Boolean saveInternalSubset)
    at System.Xml.XmlTextReaderImpl.ParseDtd()
    at System.Xml.XmlTextReaderImpl.ParseDoctypeDecl()
    at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
    at System.Xml.XmlTextReaderImpl.Read()
    at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
    at System.Xml.Linq.XDocument.Load(Stream stream, LoadOptions options)
    at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync_CreateResponse(ClientDiagnostics clientDiagnostics, Response response)
    at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri resourceUri, QueueSendMessage message, String version, Nullable`1 visibilitytimeout, Nullable`1 messageTimeToLive, Nullable`1 timeout, String requestId, Boolean async, String operationName, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageInternal(BinaryData message, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, Boolean async, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText)
    at ConsoleApp6.Program.NewSdkAsync(String connectionString, String[] messages) in C:\Users\joelv\Desktop\ConsoleApp6\ConsoleApp6\Program.cs:line 53
    ============================================================
    New SDK: ï = error
    Azure.RequestFailedException: XML specified is not syntactically valid.
    RequestId:76244a3d-b770-41af-b787-c444b4fec8ef
    Time:2021-03-27T18:28:39.4358801Z
    Status: 400 (XML specified is not syntactically valid.)
    ErrorCode: InvalidXmlDocument

    Additional Information:
    LineNumber: 1
    LinePosition: 57
    Reason: Error parsing Xml content

    Headers:
    Server: Windows-Azure-Queue/1.0,Microsoft-HTTPAPI/2.0
    x-ms-request-id: 76244a3d-b770-41af-b787-c444b4fec8ef
    x-ms-version: 2018-11-09
    x-ms-error-code: InvalidXmlDocument
    Date: Sat, 27 Mar 2021 18:28:39 GMT
    Content-Length: 327
    Content-Type: application/xml

    at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync_CreateResponse(ClientDiagnostics clientDiagnostics, Response response)
    at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri resourceUri, QueueSendMessage message, String version, Nullable`1 visibilitytimeout, Nullable`1 messageTimeToLive, Nullable`1 timeout, String requestId, Boolean async, String operationName, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageInternal(BinaryData message, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, Boolean async, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText)
    at ConsoleApp6.Program.NewSdkAsync(String connectionString, String[] messages) in C:\Users\joelv\Desktop\ConsoleApp6\ConsoleApp6\Program.cs:line 53
    ============================================================
    New SDK: ö = error
    System.Xml.XmlException: '"' is an unexpected token. Expecting whitespace. Line 1, position 50.
    at System.Xml.XmlTextReaderImpl.Throw(Exception e)
    at System.Xml.XmlTextReaderImpl.DtdParserProxy.System.Xml.IDtdParserAdapter.Throw(Exception e)
    at System.Xml.DtdParser.Throw(String res, String arg, Int32 lineNo, Int32 linePos)
    at System.Xml.DtdParser.ParseExternalId(Token idTokenType, Token declType, String& publicId, String& systemId)
    at System.Xml.DtdParser.ParseInDocumentDtd(Boolean saveInternalSubset)
    at System.Xml.DtdParser.Parse(Boolean saveInternalSubset)
    at System.Xml.DtdParser.System.Xml.IDtdParser.ParseInternalDtd(IDtdParserAdapter adapter, Boolean saveInternalSubset)
    at System.Xml.XmlTextReaderImpl.ParseDtd()
    at System.Xml.XmlTextReaderImpl.ParseDoctypeDecl()
    at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
    at System.Xml.XmlTextReaderImpl.Read()
    at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
    at System.Xml.Linq.XDocument.Load(Stream stream, LoadOptions options)
    at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync_CreateResponse(ClientDiagnostics clientDiagnostics, Response response)
    at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri resourceUri, QueueSendMessage message, String version, Nullable`1 visibilitytimeout, Nullable`1 messageTimeToLive, Nullable`1 timeout, String requestId, Boolean async, String operationName, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageInternal(BinaryData message, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, Boolean async, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText)
    at ConsoleApp6.Program.NewSdkAsync(String connectionString, String[] messages) in C:\Users\joelv\Desktop\ConsoleApp6\ConsoleApp6\Program.cs:line 53
    ============================================================
    New SDK: ü = error
    Azure.RequestFailedException: XML specified is not syntactically valid.
    RequestId:17082481-6766-45a1-937f-6f326c1c8c21
    Time:2021-03-27T18:28:39.4543823Z
    Status: 400 (XML specified is not syntactically valid.)
    ErrorCode: InvalidXmlDocument

    Additional Information:
    LineNumber: 1
    LinePosition: 57
    Reason: Error parsing Xml content

    Headers:
    Server: Windows-Azure-Queue/1.0,Microsoft-HTTPAPI/2.0
    x-ms-request-id: 17082481-6766-45a1-937f-6f326c1c8c21
    x-ms-version: 2018-11-09
    x-ms-error-code: InvalidXmlDocument
    Date: Sat, 27 Mar 2021 18:28:39 GMT
    Content-Length: 327
    Content-Type: application/xml

    at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync_CreateResponse(ClientDiagnostics clientDiagnostics, Response response)
    at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri resourceUri, QueueSendMessage message, String version, Nullable`1 visibilitytimeout, Nullable`1 messageTimeToLive, Nullable`1 timeout, String requestId, Boolean async, String operationName, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageInternal(BinaryData message, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, Boolean async, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText)
    at ConsoleApp6.Program.NewSdkAsync(String connectionString, String[] messages) in C:\Users\joelv\Desktop\ConsoleApp6\ConsoleApp6\Program.cs:line 53
    ============================================================
    New SDK: a = error
    System.Xml.XmlException: '"' is an unexpected token. Expecting whitespace. Line 1, position 50.
    at System.Xml.XmlTextReaderImpl.Throw(Exception e)
    at System.Xml.XmlTextReaderImpl.DtdParserProxy.System.Xml.IDtdParserAdapter.Throw(Exception e)
    at System.Xml.DtdParser.Throw(String res, String arg, Int32 lineNo, Int32 linePos)
    at System.Xml.DtdParser.ParseExternalId(Token idTokenType, Token declType, String& publicId, String& systemId)
    at System.Xml.DtdParser.ParseInDocumentDtd(Boolean saveInternalSubset)
    at System.Xml.DtdParser.Parse(Boolean saveInternalSubset)
    at System.Xml.DtdParser.System.Xml.IDtdParser.ParseInternalDtd(IDtdParserAdapter adapter, Boolean saveInternalSubset)
    at System.Xml.XmlTextReaderImpl.ParseDtd()
    at System.Xml.XmlTextReaderImpl.ParseDoctypeDecl()
    at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
    at System.Xml.XmlTextReaderImpl.Read()
    at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
    at System.Xml.Linq.XDocument.Load(Stream stream, LoadOptions options)
    at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync_CreateResponse(ClientDiagnostics clientDiagnostics, Response response)
    at Azure.Storage.Queues.QueueRestClient.Messages.EnqueueAsync(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri resourceUri, QueueSendMessage message, String version, Nullable`1 visibilitytimeout, Nullable`1 messageTimeToLive, Nullable`1 timeout, String requestId, Boolean async, String operationName, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageInternal(BinaryData message, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, Boolean async, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText, Nullable`1 visibilityTimeout, Nullable`1 timeToLive, CancellationToken cancellationToken)
    at Azure.Storage.Queues.QueueClient.SendMessageAsync(String messageText)
    at ConsoleApp6.Program.NewSdkAsync(String connectionString, String[] messages) in C:\Users\joelv\Desktop\ConsoleApp6\ConsoleApp6\Program.cs:line 53
    ============================================================
    New SDK: e = success
    ============================================================
    New SDK: i = success
    ============================================================
    New SDK: o = success
    ============================================================
    New SDK: u = success
    ============================================================
</details>

**Note that sending an `a` message fails after failing on a `ü` message. Sending `e` after the `a` works so it looks like the state recovers.**

**Environment:**
 - Name and version of the Library package used: Azure.Storage.Queues 12.6.0
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): Windows 10, .NET SDK 6

<details>
<summary><pre>dotnet --info</pre></summary>

    .NET SDK (reflecting any global.json):
    Version:   6.0.100-preview.1.21103.13
    Commit:    b8a03527b2

    Runtime Environment:
    OS Name:     Windows
    OS Version:  10.0.19042
    OS Platform: Windows
    RID:         win10-x64
    Base Path:   C:\Program Files\dotnet\sdk\6.0.100-preview.1.21103.13\

    Host (useful for support):
    Version: 6.0.0-preview.1.21102.12
    Commit:  9b2776d481

    .NET SDKs installed:
    5.0.100 [C:\Program Files\dotnet\sdk]
    5.0.102 [C:\Program Files\dotnet\sdk]
    5.0.200-preview.21077.7 [C:\Program Files\dotnet\sdk]
    5.0.201 [C:\Program Files\dotnet\sdk]
    6.0.100-preview.1.21103.13 [C:\Program Files\dotnet\sdk]

    .NET runtimes installed:
    Microsoft.AspNetCore.All 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
    Microsoft.AspNetCore.All 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
    Microsoft.AspNetCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
    Microsoft.AspNetCore.App 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
    Microsoft.AspNetCore.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
    Microsoft.AspNetCore.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
    Microsoft.AspNetCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
    Microsoft.AspNetCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
    Microsoft.AspNetCore.App 6.0.0-preview.1.21103.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
    Microsoft.NETCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.NETCore.App 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.NETCore.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.NETCore.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.NETCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.NETCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.NETCore.App 6.0.0-preview.1.21102.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
    Microsoft.WindowsDesktop.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.0-preview.1.21103.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

    To install additional .NET runtimes or SDKs:
    https://aka.ms/dotnet-download
</details>

 - IDE and version : VS 16.10 Preview 1.0

**Comment by joelverhagen** [comment]
### Regarding the documentation and maybe encoding my input?

By the way, I was confused by [the documentation on `SendMessageAsync`](https://docs.microsoft.com/en-us/dotnet/api/azure.storage.queues.queueclient.sendmessageasync?view=azure-dotnet#Azure_Storage_Queues_QueueClient_SendMessageAsync_System_String_):

> A message must be in a format that can be included in an XML request with UTF-8 encoding. 

Does this mean I need to XML entity encoding myself? I tried this and it seems that the SDK does it's own encoding of my `&#0000;` entities to `&amp;#0000;`.

```csharp
private static string EncodeEntities(string message)
{
    var builder = new StringBuilder(message.Length);
    for (int i = 0; i < message.Length; i += char.IsSurrogatePair(message, i) ? 2 : 1)
    {
        int c = char.ConvertToUtf32(message, i);
        if (c < ' ' || c > 126 || c == '<' || c == '&')
        {
            builder.AppendFormat("&#{0:D4};", c);
        }
        else
        {
            builder.Append((char)c);
        }
    }

    return builder.ToString();
}
`````

![image](https://user-images.githubusercontent.com/94054/112731193-11194d00-8ef3-11eb-8e6b-16da415be8b7.png)

### Nope, doesn't seem like the intended solution.

**Comment by joelverhagen** [comment]
This appears to be resolved in 12.7.0-beta.1 and 12.7.0-beta.2. Thanks!
